estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,56 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+    _order = "price DESC"
+
+    price = fields.Float()
+    status = fields.Selection(
+        selection=[("accepted", "Accepted"), ("refused", "Refused")],
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", string="Buyer", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    property_type_id = fields.Many2one("estate.property.type", related="property_id.property_type_id", store=True)
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.add(record.create_date or fields.Date.today(), days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - record.create_date.date()).days
+
+    def action_accept_offer(self):
+        if self.filtered(lambda x: x.property_id.state in ("accepted", "sold", "cancelled")):
+            raise UserError(_("This offer can't be accepted because the property is currently %s.", self.property_id.state))
+
+        self.status = "accepted"
+        self.property_id.state = "accepted"
+        self.property_id.selling_price = self.price
+        self.property_id.buyer_id = self.partner_id
+
+        return True
+
+    def action_refuse_offer(self):
+        self.status = "refused"
+        return True
+
+    _positive_offer_price = models.Constraint(
+        "CHECK(price > 0)",
+        "The offer price must be strictly positive.",
+    )
+
+    @api.model
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals["price"] <= self.env["estate.property"].browse(vals["property_id"]).best_price:
+                raise UserError(_("The offer price must be strictly greater than the current best offer."))
+            self.env["estate.property"].browse(vals["property_id"]).state = "received"
```

Comment:
same with this line, if you use an intermediate variable with your filtered property